### PR TITLE
Support descriptions for enum values in config.

### PR DESCRIPTION
**Pull request by oggy**
This goes with atom/atom#7916

### CODE REVIEW

File: spec/settings-panel-spec.coffee [hunk 1]
```diff
@@ -31,22 +31,36 @@ describe "SettingsPanel", ->
             type: 'string'
             default: 'zang'
             order: 100
+          enum:
+            title: 'An enum'
+            type: 'string'
+            default: 'one'
+            enum: [
+              ['one', 'One']
```
Comment (benogle) [inline spec/settings-panel-spec.coffee]: What about an object for clarity? `{value: 'one', description: 'One'}`

Comment (thomasjo) [inline spec/settings-panel-spec.coffee]: :+1: Makes it even clearer and sets up for making it easier to possibly adding further metadata in the future.

**Comment by thomasjo** [comment]
Specs failures seemed legit, but I kicked off a new build just to be on the safe side.

**Comment by benogle** [comment]
Likely legit cause behavior is not in master.

**Comment by oggy** [comment]
I've updated this and the other PR to use objects with `value` and `description` properties.

I think the tests are failing because the updated test here depends on the other PR being merged, and the atom on Travis would not include that. If I run locally with `ATOM_DEV_RESOURCE_PATH` pointing to the PR branch of Atom, it passes.

Is that ok? Do we need to have a test suite that works with older Atoms? I figured being a core package this was fine, but let me know if not.

**Comment by thomasjo** [comment]
> I've updated this and the other PR to use objects with value and description properties.

Awesome! :clap: 

> Is that ok? Do we need to have a test suite that works with older Atoms? I figured being a core package this was fine, but let me know if not.

I think that's totally OK. Nobody should ever install this manually anyway (outside of hacking on it), so that isn't really something we need to handle.

This looks good to me! :ship:

**Comment by Arcanemagus** [comment]
Is there a reason the associated Atom PR was merged back on 2016-01-26 but this was never merged?

**Comment by lee-dohm** [comment]
> Is there a reason the associated Atom PR was merged back on 2016-01-26 but this was never merged?

Someone made a mistake? (These things happen, y'know :laughing:)

I'm going to take a look at this and see if it can easily be updated and merged. Not sure how much drift there has been since it was last looked at.

**Comment by lee-dohm** [comment]
Closing this as it is superseded by #788.

**Comment by lee-dohm** [comment]
Thanks very much @oggy for the work you did!

**Pull request closed without merging**